gmso/tests/test_conversions.py

```diff
@@ -0,0 +1,78 @@
+from copy import deepcopy
+
+import pytest
+import sympy
+import unyt as u
+from mbuild.tests.base_test import BaseTest
+from unyt.testing import assert_allclose_units
+
+from gmso.utils.conversions import check_convert_kelvin_to_energy_units
+
+
+class TestKelvinToEnergy(BaseTest):
+    def test_K_to_kcal(self):
+        input_value = 1 * u.Kelvin / u.nm**2
+        new_value = check_convert_kelvin_to_energy_units(
+            input_value,
+            "kcal/mol",
+        )
+
+        assert new_value == u.unyt_quantity(
+            0.0019872041457050975, "kcal/(mol*nm**2)"
```

Comment:
It might be better to do the conversion manually instead of just putting the float here, so it's clear where that float comes from.
```python
import numpy as np
input_value = input_value * u.Kelvin / u.nm**2
new_value = check_convert_kelvin_to_energy_units(
    input_value,
    "kcal/mol",
)
man_conv = (
    input_value
    * u.physical_constants.boltzmann_constant 
    * u.physical_constants.avogadros_number_mks
    / u.physical_constants.avogadros_number # This is the only way to make the conversion close enough, 
    # something is weird here because this value is 0.99999993
)
assert np.isclose(new_value.to_value("kcal/mol/nm**2"), man_conv.to_value("kcal/mol/nm**2"))
```

Reply:
looks like NA is a correction for the use of 6.0221e23 rather than what is considered to be the most accurate value.  602,214,141,070,409,084,099,072
from ref:
https://www.americanscientist.org/article/an-exact-value-for-avogadros-number

seems like something that should be documented on the unyt documentation.

Reply:
I made an issue on Unyt about this Avogadros number rounding, based on @chrisiacovella 's above comment.  Hopefully, they just implement the correct value.

https://github.com/yt-project/unyt/issues/226

Reply:
The unyt group commented on the issue for Avagadros number https://github.com/yt-project/unyt/issues/226

Reply:
It looks like Unyt is going to make a more accurate Avagadros number in future releases.